### PR TITLE
add new feature: send plain json  

### PR DESCRIPTION
**Pull request by TomRt**
https://github.com/jkuri/ngx-uploader/compare/master...TomRt:master#diff-ab29e3231e988d4a2e68b8bfe2964fc5
https://github.com/jkuri/ngx-uploader/compare/master...TomRt:master#diff-63db1fd7f4b919abbc96744154f81685

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# ngx-uploader fork (version 2.2.5) with plainJson option
```
Comment (jkuri) [inline README.md]: can you remove this from readme please?

File: README.md [hunk 2]
```diff
@@ -200,6 +205,135 @@ export class AdvancedDemoComponent {
 </div>
 ```
 
+#### Advanced Example2
```
Comment (jkuri) [inline README.md]: please change title to `Advanced Example with plain JSON` or something similar.

File: README.md [hunk 3]
```diff
@@ -200,6 +205,135 @@ export class AdvancedDemoComponent {
...
+  startUploadEvent: EventEmitter<string>;
+
+  constructor(@Inject(NgZone) private zone: NgZone) {
+    this.options = new NgUploaderOptions({
+      url: 'http://api.ngx-uploader.com/upload',
+      filterExtensions: true,
+      allowedExtensions: ['txt', 'pdf'],
+      maxSize: 2097152,
+      data:{
+                '@type': "File",
```
Comment (jkuri) [inline README.md]: please fix indentation

File: README.md [hunk 4]
```diff
@@ -200,6 +205,135 @@ export class AdvancedDemoComponent {
...
+    this.startUploadEvent.emit("startUpload");
+  }
+
+  beforeUpload(ev : Event): void {
+   
+    let file: File =  ev.target['files'][0];
+    let myReader: FileReader = new FileReader();
+
+    myReader.onloadend = (e) => {
+        let tmpB64String = myReader.result.split(',');
```
Comment (jkuri) [inline README.md]: here too, use 2 spaces as indent.

File: src/services/ngx-uploader.ts [hunk 5]
```diff
@@ -30,9 +30,15 @@ export class NgUploaderService {
 
   uploadFile(file: File): void {
     let xhr = new XMLHttpRequest();
-    let payload: FormData | File;
-
-    if (this.opts.multipart) {
+    let payload: FormData | File | {};
+    
+    if (this.opts.plainJson) {
+      let plainObj = {};
```
Comment (jkuri) [inline src/services/ngx-uploader.ts]: is this needed? why you don't set `payload = JSON.stringify(this.opts.data)` directly?

File: src/services/ngx-uploader.ts [hunk 6]
```diff
@@ -30,9 +30,15 @@ export class NgUploaderService {
...
+      Object.keys(this.opts.data).forEach(k => {
+        plainObj[k] = this.opts.data[k];
+      });
+      payload = JSON.stringify(plainObj)}
```
Comment (jkuri) [inline src/services/ngx-uploader.ts]: move `}` to next line.

**Comment by TomRt** [comment]
I've fixed my modifications as proposed by jkuri

**Comment by jkuri** [comment]
@TomRt this looks alright. squashing and merging ...

**Comment by antonyRoberts** [comment]
@jkuri What is the status of this? I don't see plainJson as an option in 3.0.6 or on the master branch

**Comment by jkuri** [comment]
hi antony, looks like I missed it while refactoring. would you mind creating a PR?

**Pull request merged**